spec/classes/init_spec.rb

```diff
@@ -120,99 +121,97 @@
           it do
-            should contain_file('pam_d_login').with({
+            is_expected.to contain_file('pam_d_login').with(
               'ensure'  => 'file',
               'path'    => '/etc/pam.d/login',
               'content' => File.read(fixtures("pam_d_login.defaults.#{os}")),
               'owner'   => 'root',
               'group'   => 'root',
               'mode'    => '0644',
-            })
+            )
           end
 
           it do
-            should contain_file('pam_d_sshd').with({
+            is_expected.to contain_file('pam_d_sshd').with(
               'ensure'  => 'file',
               'path'    => '/etc/pam.d/sshd',
               'content' => File.read(fixtures("pam_d_sshd.defaults.#{os}")),
               'owner'   => 'root',
               'group'   => 'root',
               'mode'    => '0644',
-            })
+            )
           end
         else
-          it { should_not contain_file('pam_d_login') }
-          it { should_not contain_file('pam_d_sshd') }
+          it { is_expected.not_to contain_file('pam_d_login') }
+          it { is_expected.not_to contain_file('pam_d_sshd') }
         end
       end
     end
 
     context "with login_pam_access set to valid string sufficient on OS #{os}" do
       let(:facts) { v[:facts_hash] }
-      let(:params) {{ :login_pam_access => 'sufficient' }}
+      let(:params) { { login_pam_access: 'sufficient' } }
 
-      case v[:pam_d_login]
-      when 'without_pam_access'
-        it { should contain_file('pam_d_login').without_content(/account[\s]+sufficient[\s]+pam_access.so/) }
-      when 'with_pam_access'
-        it { should contain_file('pam_d_login').with_content(/^account[\s]+sufficient[\s]+pam_access.so$/) }
+      if v[:pam_d_login] == 'without_pam_access'
+        it { is_expected.to contain_file('pam_d_login').without_content(%r{account[\s]+sufficient[\s]+pam_access.so}) }
+      elsif v[:pam_d_login] == 'with_pam_access'
+        it { is_expected.to contain_file('pam_d_login').with_content(%r{^account[\s]+sufficient[\s]+pam_access.so$}) }
       else
-        it { should_not contain_file('pam_d_login') }
+        it { is_expected.not_to contain_file('pam_d_login') }
       end
     end
 
     context "with sshd_pam_access set to valid string sufficient on OS #{os}" do
       let(:facts) { v[:facts_hash] }
-      let(:params) {{ :sshd_pam_access => 'sufficient' }}
+      let(:params) { { sshd_pam_access: 'sufficient' } }
 
-      case v[:pam_d_sshd]
-      when 'without_pam_access'
-        it { should contain_file('pam_d_sshd').without_content(/account[\s]+sufficient[\s]+pam_access.so/) }
-      when 'with_pam_access'
-        it { should contain_file('pam_d_sshd').with_content(/^account[\s]+sufficient[\s]+pam_access.so$/) }
+      if v[:pam_d_sshd] == 'without_pam_access'
+        it { is_expected.to contain_file('pam_d_sshd').without_content(%r{account[\s]+sufficient[\s]+pam_access.so}) }
```

Comment:
pdk (CONVENTION): rubocop: Empty example group detected. (spec/classes/init_spec.rb:198:5)
Change statement to if to get rid of this warning. Example group is not empty for any OS. Maybe a bug report should be made to rubocp about it?